mne_bids_pipeline/steps/preprocessing/_05b_sync_eyelink.py

```diff
@@ -235,14 +245,64 @@ def sync_eyelink(
         msg = f"Saving synced data to disk."
         logger.info(**gen_log_kwargs(message=msg))
         raw.save(
-            out_files["eyelink"],
+            out_files["eyelink_eeg"],
             overwrite=True,
             split_naming="bids", # TODO: Find out if we need to add this or not
             split_size=cfg._raw_split_size, # ???
         )
         # no idea what the split stuff is...
-        _update_for_splits(out_files, "eyelink") # TODO: Find out if we need to add this or not
-
+        _update_for_splits(out_files, "eyelink_eeg") # TODO: Find out if we need to add this or not
+
+        # Extract and concatenate eye-tracking event data frames
+        et_dfs = raw_et._raw_extras[0]["dfs"]
+        df_list = [] # List to collect extracted data frames before concatenation
+
+        # Extract fixations, saccades and blinks data frames
+        for df_name, trial_type in zip(["fixations", "saccades", "blinks"], ["fixation", "saccade", "blink"]):
+            df = et_dfs[df_name]
+            df["trial_type"] = trial_type
+            df_list.append(df)
+
+        et_combined_df = pd.concat(df_list, ignore_index=True)
+        et_combined_df.rename(columns={"time":"onset"}, inplace=True)
+        et_combined_df.sort_values(by="onset", inplace=True, ignore_index=True)
+        et_combined_df = et_combined_df[ # Adapt column order
+            [
+                "onset", # needs to be first (BIDS convention)
+                "duration",
+                "end_time",
+                "trial_type",
+                "eye",
+                "fix_avg_x",
+                "fix_avg_y",
+                "fix_avg_pupil_size",
+                "sacc_start_x",
+                "sacc_start_y",
+                "sacc_end_x",
+                "sacc_end_y",
+                "sacc_visual_angle",
+                "peak_velocity"
+            ]
+        ] 
+
+        # Synchronize eye-tracking events with EEG data
+
+        # Recalculate regression coefficients (because the realign_raw function does not output them)
+        # Code snippet from `mne.preprocessing.realign_raw` function:
+        # https://github.com/mne-tools/mne-python/blob/b44c46ae7f9b6ffc5318b5d64f12906c1f2d875c/mne/preprocessing/realign.py#L69-L71
+        poly = Polynomial.fit(x=et_sync_times, y=sync_times, deg=1)
+        converted = poly.convert(domain=(-1, 1))
+        [zero_ord, first_ord] = converted.coef
+        # print(zero_ord, first_ord)
+
+        # Synchronize time stamps of ET events
+        et_combined_df["onset"] = (et_combined_df["onset"] * first_ord + zero_ord)
+        et_combined_df["end_time"] = (et_combined_df["end_time"] * first_ord + zero_ord)
+        # TODO: Do we also need to "synchronize"/adapt the duration column?
```

Comment:
```suggestion
        # TODO: to be super correct, we would need to recalculate duration column as well - but typically the slope is so close to "1" that this would typically result in <1ms differences
```